sdk/keyvault/Azure.Security.KeyVault.Secrets/MigrationGuide.md

```diff
@@ -0,0 +1,252 @@
+using (HttpClient httpClient = new HttpClient())
+{
+    SecretClientOptions options = new SecretClientOptions
+    {
+        Transport = new HttpClientTransport(httpClient)
+    };
+
+    SecretClient client = new SecretClient(
+        new Uri("https://myvault.vault.azure.net"),
+        new DefaultAzureCredential(),
+        options);
+}
+```
+
+[`ClientOptions`](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-usage-options) classes are another common feature of Azure client libraries to configure clients, including diagnostics, retry options, and transport options including your pipeline policies.
+
+### Setting secrets
+
+Previously in `Microsoft.Azure.KeyVault`, you could set a secret value using the `KeyVaultClient` and a specific Key Vault endpoint:
+
+```C# Snippet:Microsoft_Azure_KeyVault_Snippets_MigrationGuide_SetSecret
+SecretBundle secret = await client.SetSecretAsync("https://myvault.vault.azure.net", "secret-name", "secret-value");
+```
+
+Now in `Azure.Security.KeyVault.Secrets`, you set a secret value in the Key Vault you specified when constructing the `SecretClient`:
+
+```C# Snippet:Azure_Security_KeyVault_Secrets_Snippets_MigrationGuide_SetSecret
+KeyVaultSecret secret = await client.SetSecretAsync("secret-name", "secret-value");
+```
+
+Setting an existing secret in both cases will create a new version of the secret.
+
+Synchronous methods are also available on `SecretClient`, though we recommend you use asynchronous methods throughout your projects when possible for better performing applications.
+
+### Getting secrets
+
+Previously in `Microsoft.Azure.KeyVault`, you could get a secret value using the `KeyVaultClient` and a specific Key Vault endpoint:
+
+```C# Snippet:Microsoft_Azure_KeyVault_Snippets_MigrationGuide_GetSecret
+// Get the latest secret value.
+SecretBundle secret = await client.GetSecretAsync("https://myvault.vault.azure.net", "secret-name", null);
+
+// Get a specific secret value.
+SecretBundle secretVersion = await client.GetSecretAsync("https://myvault.vault.azure.net", "secret-name", "e43af03a7cbc47d4a4e9f11540186048");
+```
+
+Now in `Azure.Security.KeyVault.Secrets`, you get a secret value in the Key Vault you specified when constructing the `SecretClient`:
+
+```C# Snippet:Azure_Security_KeyVault_Secrets_Snippets_MigrationGuide_GetSecret
+// Get the latest secret value.
+KeyVaultSecret secret = await client.GetSecretAsync("secret-name");
+
+// Get a specific secret value.
+KeyVaultSecret secretVersion = await client.GetSecretAsync("secret-name", "e43af03a7cbc47d4a4e9f11540186048");
+```
+
+Synchronous methods are also available on `SecretClient`, though we recommend you use asynchronous methods throughout your projects when possible for better performing applications.
+
+### Listing secrets
+
+Previously in `Microsoft.Azure.KeyVault`, you could list secrets' properties using the `KeyVaultClient` and a specific Key Vault endpoint:
```

Comment:
This sentence does not tell the user anything more than what they can infer from the method signature. It does not set the stage for us to then tell why Track 2 is doing a better job.

Looking at the code, I would guess that the main change is around the paging.
So, we can call out here how paging works in Track 1 vs Track 2

> Previously `Microsoft.Azure.KeyVault`, you would use a combination of two methods to iterate over all the secrets. First, the `GetSecretsAsync()` method gives you a page of secrets, then you pass the `page.NextPageLink` to the `GetSecretsNextAsync()` methods to get the next page.

> Now in `Azure.Security.KeyVault.Secrets`, the `GetSecretAsync()` method returns an enumerable that you can use to go over all the secrets across all pages.

Reply:
I've changed this to:
> Now in `Azure.Security.KeyVault.Secrets`, you delete a secret in the Key Vault you specified when constructing the `SecretClient` and succinctly await or poll status on an operation to complete:

But I still think the purpose of the individual sections is to explain how to migrate the code, and not necessarily why the new way is better than the old way, which we already established, in part, at the beginning of the document.